tests/nat/eval/rag_evaluator/test_rag_evaluate.py

```
@@ -165,19 +165,29 @@ def test_ragas_to_eval_output_unexpected_entries(rag_evaluator,
+        if expected is None:
+            return actual is None
+        if isinstance(expected, float) and math.isnan(expected):
+            return isinstance(actual, float) and math.isnan(actual)
+        return actual == expected
+
     # fmt: off
     test_cases = [
-        # (scores list, expected per‑item scores list, expected average)
-        ([{metric_name: float("nan")}],            [0.0],           0.0),
-        ([{metric_name: None}],                   [0.0],           0.0),
+        # (scores list, expected per‑item scores list (preserving NaN/None), expected average (using 0.0 for NaN/None))
```

Ruff (0.14.1)

Comment contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)